crud/common/schema.lua

```diff
@@ -0,0 +1,107 @@
+local fiber = require('fiber')
+local msgpack = require('msgpack')
+local digest = require('digest')
+local vshard = require('vshard')
+
+local const = require('crud.common.const')
+
+local schema = {}
+
+local function call_reload_schema(replicasets)
+    for _, replicaset in pairs(replicasets) do
+        local master = replicaset.master
+        master.conn:reload_schema()
+    end
+end
+
+local reload_in_progress = false
+local reload_schema_cond = fiber.cond()
+
+local function reload_schema(replicasets)
+    if reload_in_progress then
+        reload_schema_cond:wait()
+    else
+        reload_in_progress = true
+
+        call_reload_schema(replicasets)
+
+        reload_schema_cond:broadcast()
+        reload_in_progress = false
+    end
+
+    return true
+end
+
+function schema.wrap_func_reload(func, ...)
+    local i = 0
+
+    local res, err, need_reload
+    while true do
+        res, err, need_reload = func(...)
+
+        if err == nil or not need_reload then
+            break
+        end
+
+        reload_schema(vshard.router.routeall())
+
+        i = i + 1
+        if i > const.RELOAD_RETRIES_NUM then
+            break
+        end
+    end
+
+    return res, err
+end
+
+local function get_space_schema_hash(space)
+    if space == nil then
+        return ''
+    end
+
+    local indexes_info = {}
+    for i = 0, table.maxn(space.index) do
+        local index = space.index[i]
+        if index ~= nil then
+            indexes_info[i] = {
+                unique = index.unique,
+                parts = index.parts,
+                id = index.id,
+                type = index.type,
+                name = index.name,
+            }
+        end
+    end
+
+    local space_info = {
+        format = space:format(),
+        indexes = indexes_info,
+    }
+
+    return digest.md5(msgpack.encode(space_info))
```

Comment:
md5 is cryptographic hash function. Please use something more lightweight.

Reply:
`digest.murmur` is used now.